src/flash/nor/km1m7xx.c

```diff
@@ -0,0 +1,588 @@
+	int					i;
+
+	uint32_t			flash_sadrs[]	= {0x00000000, 0x00000000};
+	uint32_t			flash_eadrs[]	= {0x00000000, 0x00000000};
+	uint32_t			image_sadrs		= 0x00000000;
+	uint32_t			image_eadrs		= 0x00000000;
+	int					out_flag		= 0;
+
+	if (CMD_ARGC != 1) {
+		return(ERROR_COMMAND_SYNTAX_ERROR);
+	}
+
+	/* Get all sectors information of each bank */
+	bank_num	= 0;
+	bank		= flash_bank_list();
+	while (bank != NULL) {
+
+		if (bank_num > 2) {
+			LOG_ERROR("2 bank over!!\n");
+			return(ERROR_FLASH_OPER_UNSUPPORTED);
+		}
+
+		/* Get bank information */
+		get_flash_bank_by_name(bank->name, &bank);
+
+		flash_sadrs[bank_num]	= bank->base;
+		flash_eadrs[bank_num]	= bank->base + bank->size - 1;
+
+		bank_num++;
+		bank = bank->next;
+	}
+	if (bank_num == 0) {
+		LOG_ERROR("no bank!!\n");
+		return(ERROR_FLASH_BANK_INVALID);
+	}
+
+	/* Check if the image contains data outside the flash memory area. */
+	image.base_address_set	= 0;
+	image.base_address		= 0;
+	image.start_address_set	= 0;
+
+	result = image_open(&image, CMD_ARGV[0], NULL);
+	if (result != ERROR_OK) {
+		return(result);
+	}
+
+	out_flag = 0;
+	for (i = 0; i < image.num_sections; i++) {
+
+		image_sadrs	= (uint32_t)image.sections[i].base_address;
+		image_eadrs	= (uint32_t)image.sections[i].base_address + image.sections[i].size - 1;
+
+		if (((image_sadrs < flash_sadrs[0]) || (image_eadrs > flash_eadrs[0])) &&
+			((image_sadrs < flash_sadrs[1]) || (image_eadrs > flash_eadrs[1])) ) {
+			LOG_ERROR(	"Section beyond flash memory area : "
+						"0x%08x-0x%08x\n", image_sadrs, image_eadrs);
+			out_flag = 1;
+		}
+	}
+	if (out_flag == 1) {
+		return(ERROR_FLASH_DST_OUT_OF_BANK);
+	}
+
+	return(ERROR_OK);
+}
+
+static const struct command_registration km1m7xx_subcommand_handlers[] = {
+	{
+		.name		= "erase_all_sectors",
+		.handler	= km1m7xx_handle_erase_all_sectors_command,
+		.mode		= COMMAND_EXEC,
+		.usage		= "",
+		.help		= "Erase all sectors",
+	},
+	{
+		.name		= "check_image_area",
+		.handler	= km1m7xx_handle_check_image_area_command,
+		.mode		= COMMAND_ANY,
+		.usage		= "",
+		.help		= "Check if the image contains data outside the flash memory area.",
+	},
+	COMMAND_REGISTRATION_DONE
+};
+
+static const struct command_registration km1m7xx_command_handlers[] = {
+	{
+		.name		= "km1m7xx",
+		.mode		= COMMAND_ANY,
+		.help		= "km1m7xx command group",
+		.usage		= "",
+		.chain		= km1m7xx_subcommand_handlers,
+	},
+	COMMAND_REGISTRATION_DONE
+};
+
+struct flash_driver km1m7xx_flash = {
+	.name					= "km1m7xx",
+	.usage					= "",
+	.commands				= km1m7xx_command_handlers,
+	.flash_bank_command		= km1m7xx_flash_bank_command,
+	.erase					= km1m7xx_erase,
+	.protect				= km1m7xx_protect,
+	.write					= km1m7xx_write,
+	.read					= default_flash_read,
+	.probe					= km1m7xx_probe,
+	.auto_probe				= km1m7xx_auto_probe,
+	.erase_check			= km1m7xx_erase_check,
+	.protect_check			= km1m7xx_protect_check,
+	.info					= km1m7xx_info,
+	.free_driver_priv		= default_flash_free_driver_priv,
+};
```

Comment:
Please end with a new line.

Reply:
I'm sorry, I didn't understand the comment.
The end of the file seems to be the same as other similar *.c files.

Reply:
Please refer to the following answer:
https://unix.stackexchange.com/a/18789

Reply:
The last line of km1m7xx.c is "};/n" (Binary = 7D 3B 0A). That is, it ends with a new line.
Similarly, the last line of other files (e.g. cortex_m.c) is "};/n" (Binary = 7D 3B 0A).
What's the difference?

Reply:
If km1m7xx.c ends with a new line, it should have a empty 589 line.
Maybe the GitHub review editor did not display empty 589 line correctly.

What I want looks like:
![image](https://user-images.githubusercontent.com/45277407/132615902-498e69e1-d3de-463c-8e07-bae98bdb9205.png)

Reply:
Thank you. I understand.
When opened in the same editor, the last line of the actual file has the same content.

Reply:
OK. You can ignore this comment if the last new line exists. Thank you.

Reply:
It has not changed because the end of km1m7xx.c is '\n'.